src/Command/Pull/PullCommandBase.php

```diff
@@ -507,14 +507,21 @@ private function promptChooseDatabases(
   }
 
   protected function runComposerScripts(callable $outputCallback = NULL): void {
-    if (file_exists($this->dir . '/composer.json') && $this->localMachineHelper->commandExists('composer')) {
-      $this->checklist->addItem("Installing Composer dependencies");
-      $this->composerInstall($outputCallback);
-      $this->checklist->completePreviousItem();
+    if (!file_exists(Path::join($this->dir, 'composer.json'))) {
+      $this->logger->notice('composer.json file not found. Skipping composer install.');
+      return;
```

Comment:
Negative exists logic here seems to make sense, I think.

Reply:
I like to do any validation at the start of the method, hence the negative logic. I'm open to alternatives.